state-sync/state-sync-v1/src/executor_proxy.rs

```diff
@@ -86,20 +107,69 @@ impl ExecutorProxy {
         }
     }
 
-    fn fetch_all_configs(storage: &dyn DbReader) -> Result<OnChainConfigPayload, Error> {
-        let access_paths = ON_CHAIN_CONFIG_REGISTRY
-            .iter()
-            .map(|config_id| config_id.access_path())
-            .collect();
-        let configs = storage
-            .batch_fetch_resources(access_paths)
-            .map_err(|error| {
-                Error::UnexpectedError(format!("Failed batch fetch of resources: {}", error))
-            })?;
+    fn publish_initial_on_chain_configs(
+        config_registry: &[ConfigID],
+        storage: &dyn DbReader,
+        reconfig_subscriptions: &mut Vec<ReconfigSubscription>,
+    ) -> OnChainConfigPayload {
+        let on_chain_configs = Self::fetch_all_configs(config_registry, storage)
+            .expect("[state sync] Failed initial read of on-chain configs");
+
+        for subscription in reconfig_subscriptions.iter_mut() {
+            subscription
+                .publish(on_chain_configs.clone())
+                .expect("[state sync] Failed to publish initial on-chain config");
+        }
+
+        on_chain_configs
+    }
+
+    /// Fetches the configs on-chain at the currently synchronized storage version.
+    /// Note: We cannot assume that all configs will exist on-chain. As such, we must fetch
+    /// each resource one at a time. For resources that don't exist, we'll simply log
+    /// an error. Reconfig subscribers must be able to handle on-chain configs not
+    /// existing in a reconfiguration notification.
+    fn fetch_all_configs(
+        config_registry: &[ConfigID],
+        storage: &dyn DbReader,
+    ) -> Result<OnChainConfigPayload, Error> {
         let synced_version = storage.fetch_synced_version().map_err(|error| {
             Error::UnexpectedError(format!("Failed to fetch storage synced version: {}", error))
         })?;
 
+        // Build a map from config ID to the config value found on-chain
+        let mut config_id_to_config = HashMap::new();
+        for config_id in config_registry.iter() {
+            // TODO(joshlind): refactor the move storage so we aren't batch fetching individual configs...
+            if let Ok(config_list) = storage.batch_fetch_resources(vec![config_id.access_path()]) {
```

Comment:
this is because batch fetch would fail if any of it not exist?

Reply:
Yeah, exactly. I'll refactor this code so we aren't batch fetching only one config at a time.. but for now, it'll do :)